python/lsst/daf/butler/remote_butler/_http_connection.py

```diff
@@ -0,0 +1,226 @@
+        self._client = http_client
+        self.server_url = server_url
+        self._access_token = access_token
+
+        auth_headers = get_authentication_headers(access_token)
+        headers = {"user-agent": f"RemoteButler/{__version__}"}
+
+        self._headers = auth_headers | headers
+
+    def post(self, path: str, model: BaseModel) -> httpx.Response:
+        """Send a POST request to the Butler server.
+
+        Parameters
+        ----------
+        path : `str`
+            A relative path to an endpoint.
+        model : ``BaseModel``
+            Pydantic model containing the request body to be sent to the
+            server.
+
+        Returns
+        -------
+        response: `~httpx.Response`
+            The response from the server.
+
+        Raises
+        ------
+        ButlerUserError
+            If the server explicitly returned a user-facing error response.
+        ButlerServerError
+            If there is an issue communicating with the server.
+        """
+        json = model.model_dump_json(exclude_unset=True).encode("utf-8")
+        return self._send_request(
+            "POST",
+            path,
+            content=json,
+            headers={"content-type": "application/json"},
+        )
+
+    def get(self, path: str, params: Mapping[str, str | bool] | None = None) -> httpx.Response:
+        """Send a GET request to the Butler server.
+
+        Parameters
+        ----------
+        path : `str`
+            A relative path to an endpoint.
+        params : `Mapping` [ `str` , `str` | `bool` ]
+            Pydantic model containing the request body to be sent to the
+            server.
+
+        Returns
+        -------
+        response: `~httpx.Response`
+            The response from the server.
+
+        Raises
+        ------
+        ButlerUserError
+            If the server explicitly returned a user-facing error response.
+        ButlerServerError
+            If there is an issue communicating with the server.
+        """
+        return self._send_request("GET", path, params=params)
+
+    def _get_url(self, path: str, version: str = "v1") -> str:
+        """Form the complete path to an endpoint on the server.
+
+        Parameters
+        ----------
+        path : `str`
+            The relative path to the server endpoint.
+        version : `str`, optional
+            Version string to prepend to path. Defaults to "v1".
+
+        Returns
+        -------
+        path : `str`
+            The full path to the endpoint.
+        """
+        slash = "" if self.server_url.endswith("/") else "/"
+        return f"{self.server_url}{slash}{version}/{path}"
+
+    def _send_request(
+        self,
+        method: str,
+        path: str,
+        *,
+        content: bytes | None = None,
+        params: Mapping[str, str | bool] | None = None,
+        headers: Mapping[str, str] | None = None,
+    ) -> httpx.Response:
```

Comment:
In some sense, this is the most complicated method in the file and the one that might want a couple of sentences to explain it.